app/src/main/java/org/mozilla/fenix/compose/TabUrl.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.TextUnit
+import androidx.compose.ui.unit.sp
+import mozilla.components.browser.toolbar.MAX_URI_LENGTH
+import mozilla.components.lib.publicsuffixlist.PublicSuffixList
+import org.mozilla.fenix.ext.toShortUrl
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Default layout for a tab composable caption.
+ *
+ */
+@Composable
+fun TabUrl(
+    url: String,
+    publicSuffixList: PublicSuffixList,
```

Comment:
Is there a way we can refactor this such that we don't have to worry about passing `PublicSuffixList` here?

Reply:
The only way that I could think of it's to use internally via the `context.components.publicSuffixList`, but that will make testing very hard,

Reply:
Any suggestions?

Reply:
You could make it a lambda that takes and returns a string with a reasonable default - so that it's up the consumer to provide the appropriate function:

```kotlin
@Composable
fun TabUrl(
  url: String,
  shortener: (String) -> String = { it }
  // ...
) { }

fun shortenUrl(): String {
  val publicSuffixList = context.components.publicSuffixList
  return toShortUrl(publicSuffixList).take(MAX_URI_LENGTH)
}

// Using it..
TabUrl(
  "https://veryverylongurl.com",
  ::shortenUrl,
  // ...
) { }
```

Reply:
👍🏽